cdap-api/src/main/java/io/cdap/cdap/api/artifact/UpgradedArtifact.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.artifact;
+
+import io.cdap.cdap.api.annotation.Beta;
+
+/**
+ * Describes an artifact which is upgraded for an application.
+ */
+@Beta
+public class UpgradedArtifact {
+
+  private final ArtifactId oldArtifact;
```

Comment:
nit: Artifact is a little redundant since it's already in UpgradedArtifact. We can't call it 'old' and 'new' because 'new' is a reserved java keyword. what about 'from' and 'to'?

Reply:
Removed this class as it is not useful right now.